src/include/platform/ConnectivityManager.h

```diff
@@ -171,6 +174,16 @@ class ConnectivityManager
     bool IsWiFiStationProvisioned();
     void ClearWiFiStationProvision();
     CHIP_ERROR GetAndLogWiFiStatsCounters();
+#if CHIP_DEVICE_CONFIG_ENABLE_WPA && CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
+    CHIP_ERROR SetWiFiPAFAdvertisingEnabled(bool val);
+
+    typedef void (*OnConnectionCompleteFunct)(void * appState);
+    typedef void (*OnConnectionErrorFunct)(void * appState, CHIP_ERROR err);
+    CHIP_ERROR WiFiPAFConnect(void * appState, OnConnectionCompleteFunct onSuccess, OnConnectionErrorFunct onError);
```

Comment:
The "connection" is valid only in PASE period. The connection will drop and switch to the original network connection after it's ready. These callback functions are registered through the dbus interface to wpa_supplicant that it will know if appState is going away and not report up.

Reply:
How will it know the appState is going away?  There is no way to let it know that I can see.

Reply:
We just pushed a commit which uses weak_ptr to check if the pointer is valid or not.